specs/data_structures.md

```diff
@@ -320,31 +286,178 @@ A proof into an SMT is structured as:
 
 # Erasure Coding
 
+In order to enable trust-minimized light clients (i.e. light clients that do not rely on an honest majority of validating state assumption), it is critical that light clients can determine whether the data in each block is _available_ or not, without downloading the whole block itself. The technique used here was formally described in the paper [Fraud and Data Availability Proofs: Maximising Light Client Security and Scaling Blockchains with Dishonest Majorities](https://arxiv.org/abs/1809.09044).
+
+The remainder of the subsections below specify the [2D Reed-Solomon erasure coding scheme](#2d-reed-solomon-encoding-scheme) used, along with the format of [shares](#share) and how [available data](#available-data) is arranged into shares.
+
+## Reed-Solomon Erasure Coding
+
+Reed-Solomon erasure coding is used as the underlying coding scheme. The parameters are:
+- 16-bit Galois field
+- 1:1 original:parity pieces
+- `AVAILABLE_DATA_MATRIX_HALFSIZE` original pieces (i.e `AVAILABLE_DATA_MATRIX_HALFSIZE*2` total pieces)
+- `SHARE_SIZE` bytes per piece
+
+[Leopard-RS](https://github.com/catid/leopard) is a C library that implements the above scheme with quasilinear runtime.
+
+## 2D Reed-Solomon Encoding Scheme
+
+The 2-dimensional data layout is described in this section. Data is arranged in `k * k` matrix. The roots of [NMTs](#namespace-merkle-tree) for each row and column across four quadrants of data, `Q0` to `Q3` must be computed. In other words, `2k` row roots and `2k` column roots must be computed. The row and column roots are stored in the `availableDataCommitments` of the [AvailableDataHeader](#availabledataheader).
```

Comment:
Why not talk about `AVAILABLE_DATA_MATRIX_HALFSIZE` in relation to `k` here instead of above?